script.js

```diff
@@ -1,4 +1,469 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+// ==================== Programs ====================
+// Program to generate standard deck of 52 cards. Cards have a blackjack value with Ace being either 1 or 11
+const makeDeck = () => {
+  let cardDeck = [];
+  let suits = ["♥️", "♦️", "♣️", "♠️"];
+  let rankIcons = [
+    "🅰️",
+    "2️⃣",
+    "3️⃣",
+    "4️⃣",
+    "5️⃣",
+    "6️⃣",
+    "7️⃣",
+    "8️⃣",
+    "9️⃣",
+    "🔟",
+    "🎃",
+    "👸🏻",
+    "🤴🏻",
+  ];
+
+  let suitIndex = 0;
+  while (suitIndex < suits.length) {
+    let currentSuit = suits[suitIndex];
+
+    let rankCounter = 0;
+    while (rankCounter <= 12) {
+      let cardName = rankIcons[rankCounter];
+      let blackjackValue = rankCounter + 1;
+      if (rankCounter === 0) {
+        blackjackValue = [1, 11];
+      } else if (
+        rankCounter === 10 ||
+        rankCounter === 11 ||
+        rankCounter === 12
+      ) {
+        blackjackValue = 10;
+      }
+
+      let card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        value: blackjackValue,
+      };
+
+      cardDeck.push(card);
+
+      rankCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return cardDeck;
 };
+
+// Program to shuffle cards given a particular deck
+const shuffleCards = (deck) => {
+  let i = 0;
+  while (i < deck.length) {
+    let randomIndex = getRandomIndex(deck.length);
+    let randomCard = deck[randomIndex];
+    let currentCard = deck[i];
+    deck[i] = randomCard;
+    deck[randomIndex] = currentCard;
+    i += 1;
+  }
+  return deck;
+};
+
+// Generate random index for shuffling
+const getRandomIndex = (max) => Math.floor(Math.random() * max);
+
+// Generate standard deck of 52 cards
+let deck = makeDeck();
+
+// Program to show cards
+const addCard = (card) => {
+  const div = document.getElementById("cards");
+  let p = document.createElement("p");
+  p.innerText = `${card.name} of ${card.suit}`;
+  div.appendChild(p);
+};
+
+// Program to create score
+const showScore = () => {
+  const score = document.getElementById("player-score");
+  score.innerText = `${playerScore}`;
```

Comment:
Rather than using the global variable `playerScore`, this function can be more widely used if it took a score as a parameter